sysdig/resource_sysdig_secure_scanningpolicies.go

```diff
@@ -0,0 +1,251 @@
+package sysdig
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/draios/terraform-provider-sysdig/sysdig/internal/client/secure"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceSysdigSecureScanningPolicy() *schema.Resource {
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		CreateContext: resourceSysdigScanningPolicyCreate,
+		ReadContext:   resourceSysdigScanningPolicyRead,
+		UpdateContext: resourceSysdigScanningPolicyUpdate,
+		DeleteContext: resourceSysdigScanningPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"comment": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"isdefault": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "1_0",
+			},
+			"policy_bundle_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "default",
+			},
+			"rules": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"gate": {
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateDiagFunc: validateDiagFunc(validation.StringInSlice([]string{"always", "dockerfile", "files", "licenses", "metadata", "npms", "packages", "passwd_file", "retrieved_files", "vulnerabilities", "secret_scans", "ruby_gems"}, false)),
+						},
+						"trigger": {
+							Type:     schema.TypeString,
+							Required: true,
+							// ValidateDiagFunc: TODO: create inline func to validate each trigger options depending on gate https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors
```

Comment:
There are a couple of TODOs in the new code.  Do we need to implement these now?  Or are these reminders for a later date?

Reply:
those are reminders of improvements that can be done in the future. At the end, the API validates the gates / policy parameters so this would be to "double-validate".

Reply:
those are reminders of improvements that can be done in the future. At the end, the API validates the gates / policy parameters so this would be to "double-validate".